src/commands/kv/bucket/upload.rs

```diff
@@ -94,19 +96,31 @@ fn call_put_bulk_api(
     Ok(())
 }
 
-fn filter_unchanged_remote_files(
-    pairs: Vec<KeyValuePair>,
-    exclude_keys: &HashSet<String>,
-) -> Vec<KeyValuePair> {
+fn filter_files(pairs: Vec<KeyValuePair>, already_uploaded: &HashSet<String>) -> Vec<KeyValuePair> {
     let mut filtered_pairs: Vec<KeyValuePair> = Vec::new();
     for pair in pairs {
-        if !exclude_keys.contains(&pair.key) {
+        if !already_uploaded.contains(&pair.key) && !if_ignored_prefix(&pair.key) {
             filtered_pairs.push(pair);
         }
     }
     filtered_pairs
 }
 
+// todo(gabbi): Replace all the logic below with a proper .wignore implementation
+// when possible.
+const KNOWN_UNNECESSARY_PREFIXES: &'static [&str] = &[
+    "node_modules/", // npm vendoring
+    "component---",  // Gatsby sourcemaps
```

Comment:
are these _always_ too big? curious if we should include for a first pass.

Reply:
they're apparently quite the menace (and unnecessary) for gatsby users--there are plenty of js libs that allow folks to skip generating them at all. I think we can get away with ignoring them completely, but I'd love @signalnerve's input before I do anything too rash :)

Reply:
yah word; if nothing else we can always do a one-liner with just that.

Reply:
heh just realized my comment below is on this. i would say we should opt to not do it in this one and then do the .wranglerignore to address these... thoughts?

Reply:
I'm happy to remove it for now!